Form/TableKit.swift

```diff
@@ -363,35 +363,60 @@ extension TableKit: TableAnimatable {
                                    rowIdentifier: rowIdentifier,
                                    rowNeedsUpdate: rowNeedsUpdate ?? { _, _ in true })
 
+        // Apply the updates to the currently visible rows before animating other changes (see issue #182)
+        let handledUpdates = reconfigureVisibleRowsWithChanges()
+
         view.animate(changes: changes, animation: animation)
 
-        var hasReconfiguredCells = false
-        for indexPath in view.indexPathsForVisibleRows ?? [] {
-            guard let tableIndex = TableIndex(indexPath, in: self.table) else { continue }
-            let row = table[tableIndex]
-            guard let index = from.firstIndex(where: { rowIdentifier(row) == rowIdentifier($0) }) else { continue }
+        if reconfigureVisibleRowsIfNeeded() {
+            // To to refresh cells where the cell height has changed.
+            view.beginUpdates()
+            view.endUpdates()
+        }
 
-            if let cell = view.cellForRow(at: indexPath) {
-                cell.updateBackground(forStyle: style, tableView: view, at: indexPath)
+        changesCallbacker.callAll(with: changes)
+        callbacker.callAll(with: table)
 
-                let old = from[index]
-                guard rowNeedsUpdate?(old, row) != false else {
-                    continue
+        // Returns the table indexes for which the cells were reconfigured
+        func reconfigureVisibleRowsWithChanges() -> [TableIndex] {
+            var handledUpdates: [TableIndex] = []
+            for indexPath in view.indexPathsForVisibleRows ?? [] {
+                for change in changes {
+                    if case let .row(.update(new, index)) = change,
+                       indexPath == IndexPath(row: index.row, section: index.section),
+                       let cell = view.cellForRow(at: indexPath) {
+                        let old = from[indexPath]
+                        cell.reconfigure(old: old, new: new)
+                        handledUpdates.append(index)
+                        break
```

Comment:
```suggestion
                           let old = from[indexPath]
                           cell.reconfigure(old: old, new: new)
                           handledUpdates.append(index)
                           break
```
Took a while before I realized the `if`-part was split into three rows. 😅

Reply:
That's the indentation Xcode thinks we should have :D I changed it to `guard` now and added a missing `updateBackground` call. Please take another look @enhorn .

Reply:
Ah, Xcode and it's weird indentation suggestions. 😅
Looks better with the guard. 💪

Reply:
Needs another +1 after the change 🙏

Reply:
Oh, sorry. 😅